File: server/controller/applicationController.js
const express = require('express');
const router = express.Router();
const applicationService = require('../service/applicationService');

router.get('/', async (req, res) => res.json(await applicationService.getAllApplications()));
router.get('/:id', async (req, res) => res.json(await applicationService.getApplicationById(req.params.id)));
router.get('/user/:userId', async (req, res) => res.json(await applicationService.getApplicationsByUser(req.params.userId)));
router.get('/job/:jobId', async (req, res) => res.json(await applicationService.getApplicationsByJob(req.params.jobId)));
router.post('/', async (req, res) => res.json(await applicationService.createApplication(req.body)));
router.patch('/:id/status', async (req, res) => res.json(await applicationService.updateApplicationStatus(req.params.id, req.body.status)));
router.delete('/:id', async (req, res) => res.json(await applicationService.deleteApplication(req.params.id)));

module.exports = router;

File: springboot-service/src/main/java/com/jobportal/controller/ApplicationController.java
package com.jobportal.controller;
import com.jobportal.dto.request.ApplicationRequest;
import com.jobportal.dto.response.ApplicationResponse;
import com.jobportal.service.ApplicationService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import java.util.List;
import java.util.Map;
@RestController
@RequestMapping("/api/applications")
public class ApplicationController {
    private final ApplicationService applicationService;
    @GetMapping
    public ResponseEntity<List<ApplicationResponse>> getAllApplications() {
        return ResponseEntity.ok(applicationService.getAllApplications());
    }
    @GetMapping("/{id}")
    public ResponseEntity<ApplicationResponse> getApplicationById(@PathVariable Long id) {
        return ResponseEntity.ok(applicationService.getApplicationById(id));
    }
    @GetMapping("/user/{userId}")
    public ResponseEntity<List<ApplicationResponse>> getApplicationsByUser(@PathVariable Long userId) {
        return ResponseEntity.ok(applicationService.getApplicationsByUser(userId));
    }
    @GetMapping("/job/{jobId}")
    public ResponseEntity<List<ApplicationResponse>> getApplicationsByJob(@PathVariable Long jobId) {
        return ResponseEntity.ok(applicationService.getApplicationsByJob(jobId));
    }
    @PostMapping
    public ResponseEntity<ApplicationResponse> createApplication(@Valid @RequestBody ApplicationRequest request) {
        ApplicationResponse created = applicationService.createApplication(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(created);
    }
    @PatchMapping("/{id}/status")
    public ResponseEntity<ApplicationResponse> updateStatus(@PathVariable Long id, @RequestBody Map<String, String> body) {
        String status = body.get("status");
        return ResponseEntity.ok(applicationService.updateStatus(id, status));
    }
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteApplication(@PathVariable Long id) {
        applicationService.deleteApplication(id);
        return ResponseEntity.noContent().build();
    }
    public ApplicationController(final ApplicationService applicationService) {
        this.applicationService = applicationService;
    }
}
